src/main/java/io/jenkins/plugins/pipelinegraphview/analysis/StatusAndTiming.java

```diff
@@ -0,0 +1,689 @@
+            return GenericStatus.NOT_EXECUTED;
+        }
+        boolean isLastChunk = after == null || exec.isCurrentHead(lastNode);
+        if (isLastChunk) {
+            if (run.isBuilding()) {
+                if (exec.getCurrentHeads().size() > 1
+                        && lastNode instanceof BlockEndNode) { // Check to see if all the action is on other branches
+                    BlockStartNode start = ((BlockEndNode) lastNode).getStartNode();
+                    if (start.getAction(ThreadNameAction.class) != null) {
+                        return (lastNode.getError() == null) ? GenericStatus.SUCCESS : GenericStatus.FAILURE;
+                    }
+                }
+
+                if (lastNode instanceof StepStartNode && lastNode.getAction(QueueItemAction.class) != null) {
+                    QueueItemAction.QueueState queueState = QueueItemAction.getNodeState(lastNode);
+                    if (queueState == QueueItemAction.QueueState.QUEUED) {
+                        return GenericStatus.QUEUED;
+                    } else if (queueState == QueueItemAction.QueueState.CANCELLED) {
+                        return GenericStatus.ABORTED;
+                    } else {
+                        return GenericStatus.IN_PROGRESS;
+                    }
+                }
+
+                PauseAction pauseAction = lastNode.getAction(PauseAction.class);
+                if (isPendingInput(run)
+                        && pauseAction != null
+                        && pauseAction.getCause().equals("Input")) {
+                    return GenericStatus.PAUSED_PENDING_INPUT;
+                } else {
+                    return GenericStatus.IN_PROGRESS;
+                }
+            } else {
+                // Final chunk on completed build
+                Result r = run.getResult();
+                return GenericStatus.fromResult(r);
+            }
+        }
+        ErrorAction err = lastNode.getError();
+        if (err != null) {
+            Throwable rootCause = err.getError();
+            if (rootCause instanceof FlowInterruptedException flowInterruptedException) {
+                return GenericStatus.fromResult(flowInterruptedException.getResult());
+            } else {
+                return GenericStatus.FAILURE;
+            }
+        }
+        WarningAction warning = findWorstWarningBetween(firstNode, lastNode);
+        if (warning != null) {
+            return GenericStatus.fromResult(warning.getResult());
+        }
+
+        // Previous chunk before end. If flow continued beyond this, it didn't fail.
+        return GenericStatus.SUCCESS;
+    }
+
+    @SuppressFBWarnings(
+            value = "MS_SHOULD_BE_FINAL",
+            justification = "Non-final for access from the Groovy script console")
+    public static boolean DISABLE_WARNING_ACTION_LOOKUP =
+            Boolean.getBoolean(StatusAndTiming.class.getName() + ".DISABLE_WARNING_ACTION_LOOKUP");
```

Comment:
@timja any concern with this setting changing names as a result of moving the class?

Anyone who had `org.jenkinsci.plugins.workflow.pipelinegraphanalysis.DISABLE_WARNING_ACTION_LOOKUP=true` would lose that behavior within pipeline-graph-view graphs.

They would need to set `io.jenkins.plugins.pipelinegraphview.analysis.DISABLE_WARNING_ACTION_LOOKUP=true` instead or in addition to.